tests/integration/cmd_devfile_list_test.go

```diff
@@ -143,8 +143,8 @@ var _ = Describe("odo list with devfile", func() {
 				stdout, stderr := res.Out(), res.Err()
 				Expect(stderr).To(BeEmpty())
 				Expect(helper.IsJSON(stdout)).To(BeTrue())
-				helper.JsonPathContentContain(stdout, "components.0.runningIn.#", "1")
-				helper.JsonPathContentContain(stdout, "components.0.runningIn.0", "Dev")
+				helper.JsonPathContentContain(stdout, "components.0.runningIn.dev", "true")
+				helper.JsonPathContentContain(stdout, "components.0.runningIn.deploy", "")
```

Comment:
nit: I find this check confusing. If the component is not running in Deploy mode, "deploy" will not appear in the list, why then check for it to be an empty string.

Reply:
The `JsonPathContentContain` function is not able to make a difference between is the field is not defined or if the field is empty.

This test is used to check that the field is not defined